File: balancer/service/navigate/src/protocol/protobuf/CenterStack.cc
#include "CenterStack.h"

void CenterStack::HeartbeatRsp(data::Body& body,
							   int code,
							   const std::string& info,
							   int level, 
							   int service_id, 
							   unsigned int proc_id, 
							   unsigned long long conf_update_time, 
							   unsigned int expire_second)
{
	center::CenterMsg msg;

	center::HeartbeatRsp* rsp = msg.mutable_heartbeat_rsp();
	rsp->set_level(level);
	rsp->set_service_id(service_id);
	rsp->set_proc_id(proc_id);
	rsp->set_conf_update_time(conf_update_time);
	rsp->set_expire_second(expire_second);

	::data::MsgRsp* msg_rsp = body.mutable_msg_rsp();
	msg_rsp->set_code(code);
	msg_rsp->set_info(info);

	body.mutable_service_msg()->PackFrom(msg);
}

File: balancer/service/proxy/src/protocol/protobuf/center_stack.py
# -*- coding: utf-8 -*-

from protocol.protobuf.proto_py import center_pb2


class CenterStack:

    @staticmethod
    def HeartbeatRsp(body,
                     code, info,
                     level, service_id, proc_id, conf_update_time, expire_second):
        msg = center_pb2.CenterMsg()

        msg.HeartbeatRsp.level = level
        msg.HeartbeatRsp.service_id = service_id
        msg.HeartbeatRsp.proc_id = proc_id
        msg.HeartbeatRsp.conf_update_time = conf_update_time
        msg.HeartbeatRsp.expire_second = expire_second

        body.msg_rsp.code = code
        body.msg_rsp.code = info

        body.service_msg.Pack(msg)
